CareKit/CareKit/iOS/Task/Controllers/OCKSliderTaskController.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  OCKSliderTaskController.swift
+//  
+//
+//  Created by Dylan Li on 5/26/20.
+//  Copyright © 2020 NetReconLab. All rights reserved.
+//
+
+#if !os(watchOS)
+
+import CareKitStore
+import Combine
+import Foundation
+import SwiftUI
+
+open class OCKSliderTaskController: OCKTaskController {
+    
+    /// Data used to create a `CareKitUI.SliderTaskTaskView`.
+    @Published public private(set) var viewModel: SliderTaskViewModel? {
+        willSet { objectWillChange.send() }
+    }
+    
+    /// Data used to create a `CareKitUI.SliderTaskTaskView`.
+    @Published public private(set) var value: CGFloat = 0 {
+        willSet { objectWillChange.send() }
+    }
+    
+    private var cancellable: AnyCancellable?
+    
+    public required init(storeManager: OCKSynchronizedStoreManager) {
+        super.init(storeManager: storeManager)
+        cancellable = $taskEvents.sink { [unowned self] taskEvents in
+            self.viewModel = self.makeViewModel(from: taskEvents)
+        }
+    }
+    
+    private func makeViewModel(from taskEvents: OCKTaskEvents) -> SliderTaskViewModel? {
+        guard !taskEvents.isEmpty else { return nil }
+
+        if let foundValue = taskEvents.first?.first?.outcome?.values.last?.numberValue?.doubleValue {
```

Comment:
Hi @gavirawson-apple, 
I’m not sure how to correctly pull the right value from taskEvent. For example, if an event is scheduled multiple times a day, what do I do?
Also there is a bug that could be related where when an app is restarted, the slider will default to show the preset value of the @Published var on line 24: 0. This should be a threading issue because the synchronizer gives the right value, but it only sometimes updates the view. Trying: 
DispatchQueue.main.async { [weak self] in
                self?.value = CGFloat(foundValue)
}
on line 41 does not solve the issue but perhaps something similar could.